src/integration/java/io/pinecone/helpers/AssertRetry.java

```diff
@@ -1,18 +1,21 @@
 package io.pinecone.helpers;
 
+import io.pinecone.exceptions.PineconeException;
+
 import java.io.IOException;
 import java.util.concurrent.ExecutionException;
 
 public class AssertRetry {
     private static final int maxRetry = 4;
-    private static int delay = 1500;
+    private static final int delay = 1500;
 
-    public static void assertWithRetry(AssertionRunnable assertionRunnable) throws InterruptedException {
+    public static void assertWithRetry(AssertionRunnable assertionRunnable) throws InterruptedException, PineconeException {
         assertWithRetry(assertionRunnable, 2);
     }
 
-    public static void assertWithRetry(AssertionRunnable assertionRunnable, int backOff) throws InterruptedException {
+    public static void assertWithRetry(AssertionRunnable assertionRunnable, int backOff) throws InterruptedException, PineconeException {
         int retryCount = 0;
+        int delayCount = delay;
```

Comment:
I moved to this because I noticed that if you re-used this function in the same class / test, it would continue incrementing the static `delay` value without resetting it, leading to really long wait times for certain things.